plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal.hs

```diff
@@ -309,6 +317,50 @@ gotoDefinition ideState _ msgParam = do
       isSectionArgName name (Syntax.Section _ sectionArgName _) = name == CabalFields.onelineSectionArgs sectionArgName
       isSectionArgName _ _ = False
 
+-- | CodeActions for hover messages.
+--
+-- Provides a CodeAction for displaying message on hover.
+-- If found that the filtered hover message is a dependency,
+-- adds a Documentation link.
+hover :: PluginMethodHandler IdeState LSP.Method_TextDocumentHover
+hover ide _ msgParam = do
+      nfp <- getNormalizedFilePathE uri
+      (cabalFields, _) <- runActionE "cabal.cabal-hover" ide $ useWithStaleE ParseCabalFields nfp
+      let mCursorText = CabalFields.findTextWord cursor cabalFields
+      case mCursorText of
+        Nothing ->
+          pure $ InR Null
+        Just cursorText -> do
+          (gpd, _) <- runActionE "cabal.GPD" ide $ useWithStaleE ParseCabalFile nfp
+          let depsNames = map dependencyName $ allBuildDepends $ flattenPackageDescription gpd
```

Comment:
I am not sure whether `flattenPackageDescription` is expensive, perhaps we want to cache this as well... However, fine for now, imo.